### PR TITLE
Remove completed compounding orders from backlog

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

File: spectrum-offchain/src/backlog.rs [hunk 1]
```diff
@@ -342,8 +354,30 @@ where
...
+            .iter()
+            .find(|o| o.0.order_id == ord_id)
+            .map(|w| w.0.clone());
+        if let Some(w) = w {
+            self.suspended_pq.remove(&w);
+        }
+
+        if let Some(ix) = self.revisit_queue.iter().position(|o| o.order_id == ord_id) {
+            self.revisit_queue.remove(ix);
+        }
```
Comment (oskin1) [inline spectrum-offchain/src/backlog.rs:379]: we don't need to clean queues actively. Instead, we already do it passively - in `try_pop_max_order` when `order_id` is not found in persistent store we just move to the next order.
https://github.com/spectrum-finance/spectrum-offchain-ergo/commit/46c3c26bea3ea9236422520b818ebb1131d795cb

File: spectrum-offchain/src/backlog/persistence.rs [hunk 2]
```diff
@@ -68,4 +71,25 @@ where
...
+    async fn find_orders<F>(&self, f: F) -> Vec<BacklogOrder<TOrd>>
+    where
+        F: Fn(&TOrd) -> bool + Send + 'static,
+    {
+        let db = self.db.clone();
+        spawn_blocking(move || {
+            db.iterator(rocksdb::IteratorMode::Start)
+                .filter_map(|i| {
+                    let (_, v) = i.unwrap();
+                    let b: BacklogOrder<TOrd> = bincode::deserialize(&v).unwrap();
```
Comment (oskin1) [inline spectrum-offchain/src/backlog/persistence.rs:84]: We can potentially shoot in our leg with this unsafe `.unwrap()` if we decide to store something other than `BacklogOrder<TOrd>` in the future.

File: spectrum-offchain/src/backlog.rs (hunk 1 above)
Comment (kettlebell) [reply spectrum-offchain/src/backlog.rs]: Ah yes, of course. Removed.

File: spectrum-offchain/src/backlog/persistence.rs (hunk 2 above)
Comment (kettlebell) [reply spectrum-offchain/src/backlog/persistence.rs]: Removed the `unwrap()`

**Review by oskin1 — APPROVED** [review]

**Pull request merged**